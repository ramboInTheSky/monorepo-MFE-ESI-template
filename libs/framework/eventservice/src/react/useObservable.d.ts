import { ESB } from "../esb";
export declare const useCommonObservable: (esb: ESB, callback: (data: any) => void) => void;
