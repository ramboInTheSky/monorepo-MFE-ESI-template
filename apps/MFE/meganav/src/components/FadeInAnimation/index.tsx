import React from "react"
import {Transition} from "react-transition-group"

export type FadeInAnimationProps = {
    show: boolean
    children: React.ReactNode
    timeout: number
}

export const FadeInAnimation = ({show, children, timeout}: FadeInAnimationProps) => {
    const defaultStyle = {
        transition: `opacity ${timeout}ms ease-in-out`,
        opacity: 0,
    }

    const transitionStyles = {
        entering: {opacity: 1},
        entered: {opacity: 1},
        exiting: {opacity: 0},
        exited: {opacity: 0},
    }

    return (
        <Transition in={show} timeout={timeout}>
            {state => <div style={{...defaultStyle, ...transitionStyles[state]}}>{children}</div>}
        </Transition>
    )
}

export default FadeInAnimation
