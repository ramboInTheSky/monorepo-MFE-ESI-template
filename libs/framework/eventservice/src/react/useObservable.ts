import {useEffect} from "react"
import {ESB} from "../esb"

export const useCommonObservable = (esb: ESB, callback: (data: any) => void) => {
    return useEffect(() => {
        const {subscription} = esb.subscribe(callback)
        return () => {
            subscription.unsubscribe()
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
}
