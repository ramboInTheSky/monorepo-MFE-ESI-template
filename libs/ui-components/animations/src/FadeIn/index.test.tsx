import React from "react"
import {render} from "@testing-library/react"
import {Transition} from "react-transition-group"
import FadeInAnimation from "."

jest.mock("react-transition-group", () => ({
    Transition: jest.fn(({children}) => {
        return children({})
    }),
}))

describe("FadeInAnimation", () => {
    it("should show children when in on state", () => {
        jest.useFakeTimers()
        const props = {
            show: true,
            timeout: 100,
        }
        const {asFragment, getByText} = render(
            <FadeInAnimation {...props}>
                {" "}
                <div>ciao mamma</div>{" "}
            </FadeInAnimation>,
        )
        expect(Transition).toHaveBeenCalled()
        jest.advanceTimersByTime(110)
        expect(asFragment()).toMatchSnapshot()
        expect(getByText("ciao mamma")).toBeTruthy()
    })

    it("should not show children when in off state", () => {
        const props = {
            show: false,
            timeout: 100,
        }
        const {queryByText} = render(
            <FadeInAnimation {...props}>
                {" "}
                <div>ciao mamma</div>{" "}
            </FadeInAnimation>,
        )
        expect(queryByText("ciao mamma")).not.toBeVisible()
    })
})
