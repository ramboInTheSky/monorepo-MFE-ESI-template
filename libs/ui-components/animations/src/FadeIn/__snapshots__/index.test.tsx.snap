// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInAnimation should show children when in on state 1`] = `
<DocumentFragment>
  <div
    style="transition: opacity 100ms ease-in-out; opacity: 0;"
  >
     
    <div>
      ciao mamma
    </div>
     
  </div>
</DocumentFragment>
`;
